azext_edge/edge/providers/rpsaas/adr2/command.md

```diff
@@ -0,0 +1,476 @@
+
+discovered assets are a different thing but have the same props: `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.DeviceRegistry/discoveredAssets/my-asset`
+- the discovered asset is supposed to be created from the edge/cluster rather than cloud
+- create/update/delete are still allowed via the API
+- list via the regular vs discovered should yeild different asset lists
+
+
+Main changes:
+- datapoints are now under dataset
+- addition of topic property
+- removal of some other properties
+- somehow event/dataset requirement is no longer there?
+
+### asset
+
+  create
+
+	--asset [or --name]
+	--instance
+	--resource-group
+	--endpoint-profile
+	[optional]
+	--custom-attribute
+	--data-set-file
+	--default-topic-path
+	--default-topic-retention (`keep`, `never`) [may turn this into a bool flag]
+	--description
+	--disabled [flag]
+	--display-name
+	--documentation-uri
+	--event [key=value pairs, `event_notifier`, `name` required]
+	--event-file
+	--external-asset-id
+	--hardware-revision
+	--instance-resource-group
+	--instance-subscription
+	--location
+	--manufacturer
+	--manufacturer-uri
+	--model
+	--product-code
+	--serial-number
+	--software-revision
+	--dataset-publishing-interval
+	--dataset-sampling-interval
+	--dataset-queue-size
+	--event-publishing-interval
+	--event-sampling-interval
+	--event-queue-size
+	--tags
+
+  delete
+
+	--asset [or --name]
+	--resource-group
+
+  list
+
+	[optional]
+	--discovered [bool flag]
+	--resource-group
+	--endpoint-profile
+	--instance
+	--instance-resource-group [require for instance]
+	--display-name
+	--documentation-uri
+	--external-asset-id
+	--hardware-revision
+	--instance-resource-group
+	--instance-subscription
+	--location
+	--manufacturer
+	--manufacturer-uri
+	--model
+	--product-code
+	--serial-number
+	--resource-query
+
+  show
+
+	--asset [or --name]
+	--resource-group
+
+  update
+
+	--asset [or --name]
+	--resource-group
+	[optional]
+	--custom-attribute
+	--default-topic-path
+	--default-topic-retention (`keep`, `never`) [may turn this into a bool flag]
+	--description
+	--disabled [flag]
+	--display-name
+	--documentation-uri
+	--hardware-revision
+	--instance-resource-group
+	--instance-subscription
+	--location
+	--manufacturer
+	--manufacturer-uri
+	--model
+	--product-code
+	--serial-number
+	--software-revision
+	--dataset-publishing-interval
+	--dataset-sampling-interval
+	--dataset-queue-size
+	--event-publishing-interval
+	--event-sampling-interval
+	--event-queue-size
+	--tags
+
+### asset dataset
+
+  add
+
+	--asset
+	--resource-group
+	--dataset [or --name]
+	[optional]
+	--data-point [key=value pairs, `data_source`, `name` required]
+	--data-point-file
+	--publishing-interval
+	--sampling-interval
+	--queue-size
+	--topic-path
+	--topic-retention (`keep`, `never`) [may turn this into bool flag]
```

Comment:
also thoughts on having this a flag vs keeping it as an enum? The service currently has two (retain or not) but since it is an enum - I assume they want the option of adding a third type (idk partial retain? maybe retain only specific things or newest x by time or count?)